Components/AppTabNavigator/StocksTab.js

```diff
@@ -23,16 +23,76 @@ class StocksTab extends Component {
         companyNames: [],
         companySymbols: [],
       },
+      favoritedCompanies: {
+        companyNames: [],
+        companySymbols: [],
+      },
     };
   }
 
   componentDidMount() {
-    //NetInfo.isConnected.fetch().then(isConnected => {
     if (true) {
       this.getStockCardData();
+      this.readFavorites();
     }
+
+    this.willFocusSubscription = this.props.navigation.addListener(
+      'focus',
+      () => {
+        this.readFavorites();
+        console.log("BACK BUTTON");
+      }
+    );
+
+  }
+
+  componentWillUnmount() {
+    this.willFocusSubscription.remove();
   }
 
+  componentDidUpdate(){
+    //this.readFavorites();
+  }
+
+  async readFavorites() {
+    let api_key = 'Tpk_77a598a1fa804de592413ba39f6b137a';
+    console.log('reading favorites from DB');
+    let uid = firebase.auth().currentUser.uid;
+    let favoriteRef = database().ref(`${uid}/favorites/`);
+    let favorites = [];
+     await favoriteRef.once('value', (snapshot) => {
+      snapshot.forEach(function(childSnapshot) {
+        console.log(childSnapshot.key);
+        console.log(childSnapshot.val());
+        if(childSnapshot.val() === true) {
+          favorites.push(childSnapshot.key);
+        }
+      });
```

Comment:
I think here maybe we could just change it to pushing the childSnapshot.key, since we're removing from favorites altogether instead of having AAPL:false.